app/main/routes.py

```diff
@@ -174,20 +180,83 @@ def register():
 
 @main_bp.route('/user_owner', methods=['GET', 'POST'])
 @token_auth.login_required
-def user_owner(token):
+def user_owner():
+    token = request.get_json()['token']
     user = get_one(Owner, 'token', token)
-
     if user:
         return jsonify(user.to_dict())
     return error_response(404)
 
 
 @main_bp.route('/user_sitter', methods=['GET', 'POST'])
 @token_auth.login_required
-def user_sitter(token):
+def user_sitter():
     token = request.get_json()['token']
     user = get_one(Sitter, 'token', token)
     if user:
         return jsonify(user.to_dict())
     return error_response(404)
 
+
+
+
+
+@main_bp.route('/upload_profile_image', methods=['GET', 'POST'])
+@token_auth.login_required
+def upload_file():
+    file = request.files['file']
+    file.filename = secure_filename(file.filename)
+    output   	  = upload_file_to_s3(file, current_app.config["S3_BUCKET"])
+    token = request.headers['Authorization'].split(" ")[1]
+    collection = Sitter if Sitter.check_token(token) else Owner
+    u = collection.check_token(token)
+    u.profile_image = str(output)
+    u.save()
+    return jsonify(u.to_dict(include_email=True))
+
+
+def upload_file_to_s3(file, bucket_name, acl="public-read"):
+    s3 = boto3.client(
+        "s3",
+        region_name='ca-central-1',
+        aws_access_key_id=current_app.config["S3_KEY"],
+        aws_secret_access_key=current_app.config["S3_SECRET"]
+    )
+    try:
+        s3.upload_fileobj(
+            file,
+            bucket_name,
+            file.filename,
+            ExtraArgs={
+                "ACL": acl,
+                "ContentType": file.content_type
+            }
+        )
+    except Exception as e:
+        # This is a catch all exception, edit this part to fit your needs.
+        print("Something Happened: ", e)
```

Comment:
lolll... definitely want to use a logger here

Reply:
will do it